### PR TITLE
Implement ListPoolAttesterSlashings in the beacon API

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

What does this PR do? Why is it needed?

This PR implements beacon API's `ListPoolAttesterSlashings` method according to the spec: https://ethereum.github.io/eth2.0-APIs/#/Beacon/getPoolAttesterSlashings

Which issues(s) does this PR fix?

Part of #7510

Other notes for review

N/A

### CODE REVIEW

File: beacon-chain/rpc/beaconv1/pool.go [hunk 1]
```diff
@@ -56,3 +78,24 @@ func (bs *Server) ListPoolVoluntaryExits(ctx context.Context, req *ptypes.Empty)
 func (bs *Server) SubmitVoluntaryExit(ctx context.Context, req *ethpb.SignedVoluntaryExit) (*ptypes.Empty, error) {
 	return nil, errors.New("unimplemented")
 }
+
+func mapAttestation(attestation *eth.IndexedAttestation) *ethpb.IndexedAttestation {
+	a := &ethpb.IndexedAttestation{
+		AttestingIndices: attestation.AttestingIndices,
```
Comment (rauljordan) [inline beacon-chain/rpc/beaconv1/pool.go:84]: Any chance there could be a bad attestation that makes it into this and causes the code to panic?

Comment (prestonvanloon) [reply beacon-chain/rpc/beaconv1/pool.go:84]: +1 

```go
if attestation == nil {
  return nil
}
```

File: beacon-chain/rpc/beaconv1/pool.go [hunk 2]
```diff
@@ -56,3 +78,24 @@ func (bs *Server) ListPoolVoluntaryExits(ctx context.Context, req *ptypes.Empty)
 func (bs *Server) SubmitVoluntaryExit(ctx context.Context, req *ethpb.SignedVoluntaryExit) (*ptypes.Empty, error) {
 	return nil, errors.New("unimplemented")
 }
+
+func mapAttestation(attestation *eth.IndexedAttestation) *ethpb.IndexedAttestation {
```
Comment (terencechain) [inline beacon-chain/rpc/beaconv1/pool.go:82]: What does `mapAttestation` mean? Do you just mean copy?

Comment (rkapka) [reply beacon-chain/rpc/beaconv1/pool.go:82]: These are different types - `*eth` and `*ethpb`. It's a mapping from old proto to new proto. I will change the name to make it more clear.

Comment (terencechain) [reply beacon-chain/rpc/beaconv1/pool.go:82]: Sorry i missed the prefix. This also seems useful, perhaps a better file location in `pool` or even `shared`

Comment (prestonvanloon) [reply beacon-chain/rpc/beaconv1/pool.go:82]: This is a temporary translation between the v1alpha1 and v1 objects. A comment would be helpful here.

Comment (0xKiwi) [reply beacon-chain/rpc/beaconv1/pool.go:82]: In https://github.com/prysmaticlabs/prysm/pull/7525/ I made a migration package for utils like this, maybe that would be useful here?

File: beacon-chain/rpc/beaconv1/pool.go (hunk 1 above)
Comment (rkapka) [reply beacon-chain/rpc/beaconv1/pool.go]: I don't think this is possible. A bad attestation would never result in a slashing, right?

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**